src/Generators/Generator.php

```diff
@@ -312,98 +312,45 @@ public function getTemplatePath()
      * relative to [[templatePath]].
      * @param array $params list of parameters to be passed to the template file.
      * @return string the generated code
+     * @throws ReflectionException
+     * @throws Throwable
+     * @throws ViewNotFoundException
      */
-    public function render($template, $params = [])
+    public function render($template, $params = []): string
     {
-        $view = new View($this->app, new Theme());
         $params['generator'] = $this;
 
-        return $view->renderFile($this->getTemplatePath() . '/' . $template, $params, $this);
-    }
-
-    /**
-     * Validates the template selection.
-     * This method validates whether the user selects an existing template
-     * and the template contains all required template files as specified in [[requiredTemplates()]].
-     */
-    public function validateTemplate()
-    {
-        $templates = $this->templates;
-
-        if (!isset($templates[$this->template])) {
-            $this->addError('template', 'Invalid template selection.');
-        } else {
-            $templatePath = $this->templates[$this->template];
-            foreach ($this->requiredTemplates() as $template) {
-                if (!is_file(Yii::getAlias($templatePath . '/' . $template))) {
-                    $this->addError('template', "Unable to find the required code template file '$template'.");
-                }
-            }
-        }
+        return $this->renderTemplate($this->getTemplatePath() . '/' . $template, $params);
     }
 
-    /**
-     * An inline validator that checks if the attribute value refers to an existing class name.
-     * If the `extends` option is specified, it will also check if the class is a child class
-     * of the class represented by the `extends` option.
-     * @param string $attribute the attribute being validated
-     * @param array $params the validation options
-     */
-    public function validateClass($attribute, $params)
+    protected function renderTemplate(string $template, array $params): string
```

Comment:
Can we use [`View::render`](https://github.com/yiisoft/view/blob/master/src/View.php#L227) instead?